### PR TITLE
Add possibility to filter the collision BCs and merge digits in adjacent BCs

### PR DESCRIPTION
**Pull request by dstocco**
The filtering is not activated by default. This possibility is mainly useful for testing purposes.
The commits are meant to be atomic: please do not squash

### CODE REVIEW

**Comment by dstocco** [comment]
@aphecetche : errors seem unrelated. I guess you can safely start the review

File: Detectors/MUON/MID/Workflow/README.md [hunk 1]
```diff
@@ -117,6 +117,32 @@ o2-mid-reco-workflow --change-local-to-BC <value>
...
 
+## MID BC filtering
+
+The MID time resolution is better than 25 ns, allowing it to distinguish between different BCs.
+However, a spread of the signal was observed, probably due an insufficient equalization of the delays across the various detection element.
+In order to study and possibly correct for any inefficiency arising from this spread, it is possible to select the BCs corresponding to a collision and merge the digits in a configurable window around this BC.
+To this aim, it is enough to run the reconstruction with the option:
+
+```bash
+o2-mid-reco-workflow --filter-BC <value>
```
Comment (aphecetche) [inline Detectors/MUON/MID/Workflow/README.md]: actually the filter-BC being a bool I think you can’t specify a value, can you ? 
also, seems most of the boolean options we have in O2 follow the naming convention `--enable-xxx` or `disable-xxx`, so I would follow it here, e.g. `--enable-filter-bc` (as the default is false).

File: Detectors/MUON/MID/Workflow/src/FilteringBCSpec.cxx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    mFiltererBC.setBCDiffHigh(FiltererBCParam::Instance().maxBCDiffHigh);
+    mFiltererBC.setSelectOnly(FiltererBCParam::Instance().selectOnly);
+  }
+
+  void finaliseCCDB(of::ConcreteDataMatcher matcher, void* obj)
+  {
+    if (o2::base::GRPGeomHelper::instance().finaliseCCDB(matcher, obj)) {
+      LOG(info) << "Update Bunch Filling for MID";
+      mFiltererBC.setBunchFilling(o2::base::GRPGeomHelper::instance().getGRPLHCIF()->getBunchFilling());
+      return;
```
Comment (aphecetche) [inline Detectors/MUON/MID/Workflow/src/FilteringBCSpec.cxx]: that's a detail but the return is superfluous

File: Detectors/MUON/MID/Workflow/src/FilteringBCSpec.cxx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  }
+
+  void run(of::ProcessingContext& pc)
+  {
+    updateTimeDependentParams(pc);
+
+    auto data = specs::getData(pc, "mid_filter_BC_in", EventType::Standard);
+    auto inROFRecords = specs::getRofs(pc, "mid_filter_BC_in", EventType::Standard);
+
+    std::unique_ptr<const o2::dataformats::MCTruthContainer<MCLabel>> inMCContainer = mUseMC ? specs::getLabels(pc, "mid_filter_BC_in") : nullptr;
```
Comment (aphecetche) [inline Detectors/MUON/MID/Workflow/src/FilteringBCSpec.cxx]: any particular reason you're not using the simpler `auto inMCContainer = ...` like for the other inputs ?

File: Detectors/MUON/MID/Filtering/src/FiltererBC.cxx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  // First we check the simple case to save time
+  if (mBunchFilling.testInteractingBC(bc)) {
+    return bc;
+  }
+
+  // Then we check for neighbor BCs
+  // We split the check in two so that we start the check from the input bc
+  // and we move farther from it.
+
+  int matchedBC = 3 * o2::constants::lhc::LHCMaxBunches;
```
Comment (aphecetche) [inline Detectors/MUON/MID/Filtering/src/FiltererBC.cxx]: why `3*` ?

Comment (dstocco) [reply Detectors/MUON/MID/Filtering/src/FiltererBC.cxx]: I just need a value larger than o2::constants::lhc::LHCMaxBunches. The factor does not really matter.

Comment (dstocco) [reply Detectors/MUON/MID/Filtering/src/FiltererBC.cxx]: Changed into: `int matchedBC = std::numeric_limits<int>::max();`

**Review by aphecetche — APPROVED** [review]

**Pull request merged**